static/js/pages/instances/forms/SnapshotsForm.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { FC, ReactNode, useState } from "react";
+import {
+  Col,
+  Input,
+  RadioInput,
+  Row,
+  Select,
+} from "@canonical/react-components";
+import { booleanFields } from "util/instanceOptions";
+import {
+  SharedFormikTypes,
+  SharedFormTypes,
+} from "pages/instances/forms/sharedFormTypes";
+
+export interface SnapshotFormValues {
+  snapshots_pattern?: string;
+  snapshots_expiry?: string;
+  snapshots_schedule?: string;
+  snapshots_schedule_stopped?: string;
+}
+
+export const snapshotsPayload = (values: SharedFormTypes) => {
+  return {
+    ["snapshots.pattern"]: values.snapshots_pattern,
+    ["snapshots.schedule.stopped"]: values.snapshots_schedule_stopped,
+    ["snapshots.schedule"]: values.snapshots_schedule,
+    ["snapshots.expiry"]: values.snapshots_expiry,
+  };
+};
+
+interface Props {
+  formik: SharedFormikTypes;
+  children?: ReactNode;
+}
+
+const SnapshotsForm: FC<Props> = ({ formik, children }) => {
+  const [cronSyntax, setCronSyntax] = useState(true);
+  const isInstance = formik.values.type === "instance";
+
+  return (
+    <>
+      {children}
+      <Row>
+        <Col size={8}>
+          <Input
+            label="Snapshot name pattern"
+            placeholder="Enter name pattern"
+            help={
+              <>
+                Pongo2 template string that represents the snapshot name (used
+                for scheduled snapshots and unnamed snapshots) see{" "}
+                <a
+                  className="p-link--external"
+                  href="https://linuxcontainers.org/lxd/docs/latest/reference/instance_options/#instance-options-snapshots-names"
+                  target="_blank"
+                  rel="noreferrer"
+                >
+                  Automatic snapshot names
+                </a>
+              </>
+            }
+            name="snapshots_pattern"
+            type="text"
+            value={formik.values.snapshots_pattern}
+            onBlur={formik.handleBlur}
+            onChange={formik.handleChange}
+          />
+          <Input
+            label="Expire after"
+            name="snapshots_expiry"
+            help="Controls when snapshots are to be deleted (expects an expression like 1M 2H 3d 4w 5m 6y)"
+            type="text"
+            value={formik.values.snapshots_expiry}
+            onBlur={formik.handleBlur}
+            onChange={formik.handleChange}
+          />
+          <Select
+            label="Snapshot stopped instances"
+            name="snapshots_schedule_stopped"
+            onBlur={formik.handleBlur}
+            onChange={formik.handleChange}
+            options={booleanFields(isInstance)}
+            value={formik.values.snapshots_schedule_stopped}
+          />
+          <hr />
+          <div className="snapshot-schedule">
+            <RadioInput
+              labelClassName="right-margin"
+              label="Cron syntax"
+              checked={cronSyntax}
+              onChange={() => setCronSyntax(true)}
```

Comment:
Why not setting back the value to the empty string? I find this behaviour confusing:

![Peek 2023-03-07 12-43](https://user-images.githubusercontent.com/56583786/223412982-cf5707ce-e4e7-4ee1-8bb6-c47b4d75ad94.gif)

1) "Cron syntax" selected by default, with no default value (empty string).
2) I switch to "Manual configuration", every minute is selected by default.
3) I switch back to "Cron syntax" and I find a value filled in that is equivalent to every minute.

Is this a desired behaviour? Or is this just confusing the user? (also for @al-bakalova @yurii-vasyliev )

Reply:
Right, clearing the value is a better approach here. Also, the default for manual would better be `@daily`.